src/Strategy/TypeCastStrategy.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-hydrator for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-hydrator/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-hydrator/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Hydrator\Strategy;
+
+use Laminas\Hydrator\Exception\InvalidArgumentException;
+
+/**
+ * @template T
+ */
+final class TypeCastStrategy implements StrategyInterface
+{
+    private const TYPE_INT = 'int';
+    private const TYPE_FLOAT = 'float';
+    private const TYPE_STRING = 'str';
+
+    /** @var string */
+    private $type;
+
+    /**
+     * @return self<int>
+     */
+    public static function createToInt(): self
+    {
+        return new self(self::TYPE_INT);
+    }
+
+    /**
+     * @return self<float>
+     */
+    public static function createToFloat(): self
+    {
+        return new self(self::TYPE_FLOAT);
+    }
+
+    /**
+     * @return self<string>
+     */
+    public static function createToString(): self
+    {
+        return new self(self::TYPE_STRING);
+    }
```

Comment:
These methods are not relevant for this hydrator strategy. Please remove them.

Reply:
Mh, but this would result in something like `new TypeCastStrategy(TypeCaseStrategy::TYPE_STRING)` to create a strategy for a specific type. Or is there another way to create/configure strategies that I'm not aware of?

Reply:
Ah, I missed that the visibility of the constructor is marked as private.